datastore/tests/unit/test_transaction.py

```diff
@@ -212,6 +218,95 @@ class Foo(Exception):
+        client = _Client(project, datastore_api=ds_api)
+        xact = self._make_one(client)
+        self.assertTrue(isinstance(xact.options, self._get_options_class()))
+
+    def test_read_only(self):
+        project = 'PROJECT'
+        id_ = 850302
+        ds_api = _make_datastore_api(xact=id_)
+        client = _Client(project, datastore_api=ds_api)
+        read_only = self._get_options_class().ReadOnly()
+        options = self._make_options(read_only=read_only)
+        xact = self._make_one(client, read_only=True)
+        xact.begin()
+        begin = ds_api.begin_transaction
+        begin.assert_called_once()
+        self.assertEqual(begin.call_count, 1)
+
+    def test_read_write(self):
+        project = 'PROJECT'
+        id_ = 850304
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options = self._make_options(read_write=
+                                     self._get_options_class().ReadWrite())
+        xact = self._make_one(client)
+        xact.begin()
+        begin = ds_api.begin_transaction
+        begin.assert_called_once()
+        self.assertEqual(begin.call_count, 1)
+
+    def test_previous_tid(self):
+        project = 'PROJECT'
+        id_ = 943243
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options_klass = self._get_options_class()
+        read_write = options_klass.ReadWrite(previous_transaction=b'321')
+        options = self._make_options(read_write=read_write)
+        xact = self._make_one(client)
+        xact.begin()
+        begin = ds_api.begin_transaction
+        begin.assert_called_once()
+        self.assertEqual(begin.call_count, 1)
+
+    def test_put_read_only(self):
+        project = 'PROJECT'
+        id_ = 943243
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options_klass = self._get_options_class()
+        options = self._make_options(read_only=options_klass.ReadOnly())
+        entity = _Entity()
+        xact = self._make_one(client, read_only=True)
+        xact.begin()
+        with self.assertRaises(RuntimeError):
+            xact.put(entity)
+
+    def test_put_read_write(self):
+        project = 'PROJECT'
+        id_ = 943243
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options_klass = self._get_options_class()
+        read_write = options_klass.ReadWrite(previous_transaction=b'321')
+        options = self._make_options(read_write=read_write)
+        entity = _Entity()
+        xact = self._make_one(client)
+        xact.begin()
+        xact.put(entity)
+        mutated_entity = _mutated_pb(self, xact.mutations, 'insert')
+        self.assertEqual(mutated_entity.key, entity.key.to_protobuf())
+
+
+def _mutated_pb(test_case, mutation_pb_list, mutation_type):
```

Comment:
Please remove this helper (only used by `test_put_read_write`, which should be removed)